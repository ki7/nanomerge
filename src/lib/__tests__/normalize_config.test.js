import normalizeConfig from "../normalize_config";

it("Must be function", () => {
  expect(typeof normalizeConfig).toBe("function");
});

it("Must return default config", () => {
  expect(normalizeConfig({})).toMatchSnapshot();
});

it("Must set user config", () => {
  const config = {
    strategy: {
      array: "replace"
    },

    types: {
      mode: "replace",
      list: [{}]
    }
  };

  expect(normalizeConfig(config)).toMatchSnapshot();
});

describe("Must validate configuration", () => {
  const cases = {
    empty: {},
    types: {
      types: {
        mode: "replace",
        list: [{}, {}]
      }
    },
    strategy: {
      strategy: {
        array: "replace"
      }
    },
    force: {
      force: true
    }
  };

  Object.keys(cases).forEach(testCaseName => {
    it(`Must set ${testCaseName}`, () => {
      expect(normalizeConfig(cases[testCaseName])).toMatchSnapshot();
    });
  });
});
