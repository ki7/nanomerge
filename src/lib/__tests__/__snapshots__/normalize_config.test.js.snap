// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Must return default config 1`] = `
Object {
  "force": false,
  "strategy": Object {},
  "types": Object {
    "list": Array [],
    "mode": "add",
  },
}
`;

exports[`Must set user config 1`] = `
Object {
  "force": false,
  "strategy": Object {
    "array": "replace",
  },
  "types": Object {
    "list": Array [
      Object {},
    ],
    "mode": "replace",
  },
}
`;

exports[`Must validate configuration Must set empty 1`] = `
Object {
  "force": false,
  "strategy": Object {},
  "types": Object {
    "list": Array [],
    "mode": "add",
  },
}
`;

exports[`Must validate configuration Must set force 1`] = `
Object {
  "force": true,
  "strategy": Object {},
  "types": Object {
    "list": Array [],
    "mode": "add",
  },
}
`;

exports[`Must validate configuration Must set strategy 1`] = `
Object {
  "force": false,
  "strategy": Object {
    "array": "replace",
  },
  "types": Object {
    "list": Array [],
    "mode": "add",
  },
}
`;

exports[`Must validate configuration Must set types 1`] = `
Object {
  "force": false,
  "strategy": Object {},
  "types": Object {
    "list": Array [
      Object {},
      Object {},
    ],
    "mode": "replace",
  },
}
`;
